app/controllers/collection_drafts_controller.rb

```diff
@@ -89,6 +87,11 @@ def update
       when 'Save'
         # tried to use render to avoid another request, but could not get form name in url even with passing in location
         get_resource_form = params['next_section']
+        if get_resource_form == 'descriptive_keywords' &&  get_resource.gkr_logging_active?
```

Comment:
This is only capturing the 'save' action.  All of this tree is actually saving the draft.  This should be moved into the GKR block above.

Reply:
moved the block out of the "when 'Save'" and into the main routine.

Reply:
agree